### ISSUE TITLE
Unable to handle large audio,OutOfMemory

### ISSUE THREAD

**Issue by FantasyGmm** [Zeugma440/atldotnet#196]
## The problem

I use ATL to batch process song tags, and in most cases, it works normally until it encounters a large lossless file (such as 96KHz24Bit 110MB), which it cannot read and prompts' System. OutOfMemoryException '
Some tags can be read, but some custom tags cannot be read. When I try to read them, I will output some garbled code, and the program has crashed

## Environment

* ATL version (or git revision) that exhibits the issue: 4.32.0
* OS/version used to run ATL: Windows 11 22H2
* Use .Net 7.0
## Details
Console Log
```
Exception of type 'System.OutOfMemoryException' was thrown.
   at System.String.Concat(String str0, String str1, String str2)
   at ATL.AudioData.IO.VorbisTag.SetMetaField(String ID, String data, Boolean readAllMetaFrames, String zone, Byte tagVersion, UInt16 streamNumber, String language)
   at ATL.AudioData.IO.VorbisTag.read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.MetaDataIO.Read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.FLAC.Read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.FLAC.Read(Stream source, SizeInfo sizeInfo, ReadTagParams readTagParams)
   at ATL.AudioData.AudioDataManager.read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.AudioDataManager.read(Stream source, Boolean readEmbeddedPictures, Boolean readAllMetaFrames, Boolean prepareForWriting)
   at ATL.AudioData.AudioDataManager.ReadFromFile(Boolean readEmbeddedPictures, Boolean readAllMetaFrames)
Exception of type 'System.OutOfMemoryException' was thrown.
   at System.String.Concat(String str0, String str1, String str2)
   at ATL.AudioData.IO.VorbisTag.SetMetaField(String ID, String data, Boolean readAllMetaFrames, String zone, Byte tagVersion, UInt16 streamNumber, String language)
   at ATL.AudioData.IO.VorbisTag.read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.MetaDataIO.Read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.FLAC.Read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.IO.FLAC.Read(Stream source, SizeInfo sizeInfo, ReadTagParams readTagParams)
   at ATL.AudioData.AudioDataManager.read(Stream source, ReadTagParams readTagParams)
   at ATL.AudioData.AudioDataManager.read(Stream source, Boolean readEmbeddedPictures, Boolean readAllMetaFrames, Boolean prepareForWriting)
   at ATL.AudioData.AudioDataManager.ReadFromFile(Boolean readEmbeddedPictures, Boolean readAllMetaFrames)
[2023-05-07 21:39:39.299] - 0022088529-Title : Gravity
[2023-05-07 21:39:39.299] - 0022088529-Artist : IRyS
[2023-05-07 21:39:39.300] - 0022088529-Album : Gravity
[2023-05-07 21:39:39.300] - 0022088529-Recording year : 0
[2023-05-07 21:39:39.300] - 0022088529-Track number : 1
[2023-05-07 21:39:39.301] - 0022088529-Disc number : 1
[2023-05-07 21:39:39.301] - 0022088529-Comment : JASRAC / cover corp.
[2023-05-07 21:39:39.301] - 0022088529-Duration (s) : 202
[2023-05-07 21:39:39.301] - 0022088529-Bitrate (KBps) : 4642
[2023-05-07 21:39:39.301] - 0022088529-Number of channels : 2
[2023-05-07 21:39:39.301] - 0022088529-Channels arrangement : Stereo (2/0.0)
[2023-05-07 21:39:39.302] - 0022088529-Has variable bitrate audio : no
[2023-05-07 21:39:39.302] - 0022088529-Has lossless audio : yes
```

## Code To Reproduce Issue [ Good To Have ]

```
        var theTrack = new Track(url);
        var fileName = Path.GetFileNameWithoutExtension(url);
        if (Config.DEBUG_LOG && !isBatch)
        {
            LogDebug($"{fileName}-Title : {theTrack.Title}");
            LogDebug($"{fileName}-Artist : {theTrack.Artist}");
            LogDebug($"{fileName}-Album : {theTrack.Album}");
            LogDebug($"{fileName}-Recording year : {theTrack.Year}");
            LogDebug($"{fileName}-Track number : {theTrack.TrackNumber}");
            LogDebug($"{fileName}-Disc number : {theTrack.DiscNumber}");
            LogDebug($"{fileName}-Comment : {theTrack.Comment}");
            LogDebug($"{fileName}-Duration (s) : {theTrack.Duration}");
            LogDebug($"{fileName}-Bitrate (KBps) : {theTrack.Bitrate}");
            LogDebug($"{fileName}-Number of channels : {theTrack.ChannelsArrangement.NbChannels}");
            LogDebug($"{fileName}-Channels arrangement : {theTrack.ChannelsArrangement.Description}");
            LogDebug($"{fileName}-Has variable bitrate audio : {(theTrack.IsVBR ? "yes" : "no")}");
            LogDebug($"{fileName}-Has lossless audio : {(AudioDataIOFactory.CF_LOSSLESS == theTrack.CodecFamily ? "yes" : "no")}");
        
            foreach (KeyValuePair<string, string> field in theTrack.AdditionalFields)
            {
                LogDebug($"{fileName}-Custom field {field.Key} : value = {field.Value}");
            }
        }
```

**Comment by Zeugma440** [comment]
Hi and thanks for your feedback.

ATL can process files much larger than 110MB. There's probably an issue with the file you're using.

Could you send it to me using a 3rd party cloud (e.g. Google Drive, Mega, DropBox...) ?

Thanks in advance~

**Comment by FantasyGmm** [comment]
It may be a problem with Mora's tag, but the source file contains user info,I used FB2K to change the numbers to the same, but the number of digits did not change
[This is the software I wrote using this library](https://github.com/FantasyGmm/MoraAudioModify)
[This is the music I bought from Mora,After closing the issue, I will delete it](https://alumnivtcedu-my.sharepoint.com/:u:/g/personal/sorachankc_alumni_vtc_edu_hk/EUjEExAIP4hPgtrmjtd5KGUBX0y_KJv5CXIyfVpHs7ZZ6w?e=Y1Zdqn)

**Comment by Zeugma440** [comment]
Unfortunately, I can't reproduce your issue with the code and the file you provided. Everything runs fine and the debugger shows 12MB max memory usage.

![image](https://user-images.githubusercontent.com/29414549/236695460-1a671f49-6213-400b-8237-c68084ff99d8.png)


Are you sure you provided me with the correct file?

What's more, the link you provided to https://github.com/FantasyGmm/MoraAudioModify doesn't work. Is it a private repo?

**Comment by FantasyGmm** [comment]
Sorry, I forgot he's a private. I'll make it public instead
The file modified using FB2K can be read normally, but if it is a downloaded source file, it cannot be read. Can you provide me with an email so that I can send the source file to you via email
This is when I tried to load the audio, I made a breakpoint on line 127 of Program.cs
![img](https://user-images.githubusercontent.com/16450052/236733428-18d078ed-e33c-4ed4-93c0-727238bef972.png)

**Comment by Zeugma440** [comment]
> Can you provide me with an email so that I can send the source file to you via email

You can contact me at [my github username] at [not-cold] mail [dot] com

**Comment by FantasyGmm** [comment]
Okay, I sent it to you

**Comment by Zeugma440** [comment]
Thanks for your help. The problem has been fixed and published in today's v4.33.

**Comment by FantasyGmm** [comment]
Thank you for your code
